### ISSUE TITLE
Missing cookie in oauth workflow

### ISSUE THREAD

**Issue by kardeiz** [SergioBenitez/Rocket#567]
**Rocket 0.3.6 on CentOS 7**

I have a strange issue where cookies are not being read after a series of redirects (potentially a bug).

Here is a minimal test case (note: I am not actually using my access token like this in my real project):

```rust
#![feature(plugin, custom_derive)]
#![plugin(rocket_codegen)]

extern crate rocket;
extern crate oauth2;

use rocket::response::content;
use rocket::response::Redirect;
use rocket::http::{Cookie, Cookies};

#[derive(FromForm)]
struct CodeWrapper { code: String }

fn facebook_config() -> ::oauth2::Config {
    let mut oauth = ::oauth2::Config::new(
        ::std::env::var("FACEBOOK_APP_ID").unwrap().as_str(), 
        ::std::env::var("FACEBOOK_APP_SECRET").unwrap().as_str(), 
        "https://www.facebook.com/v2.12/dialog/oauth", 
        "https://graph.facebook.com/v2.12/oauth/access_token");
    oauth = oauth
        .set_redirect_url("http://localhost:3000/oauth/fb/handle")
        .add_scope("public_profile,email");
    oauth
}

#[get("/")]
fn index(cookies: Cookies) -> content::Html<String> {
    println!("{:?}", &cookies);
    content::Html(format!("<a href=\"{}\">Sign in</a>", facebook_config().authorize_url()))
}

#[get("/oauth/fb/handle?<code_wrapper>")]
fn oauth_fb_handle(mut cookies: Cookies, code_wrapper: CodeWrapper) -> Redirect {
    let token = facebook_config().exchange_code(code_wrapper.code).unwrap();
    cookies.add_private(Cookie::new("token", token.access_token));
    Redirect::to("/")
}

fn main() {
    rocket::ignite().mount("/", routes![index, oauth_fb_handle]).launch();
}
```

When I click the link to sign in with Facebook, and then authorize my app with Facebook, I am redirected to my index route as expected. However, my `token` cookie is not printed in my console. 

If I refresh the page, then my cookie is printed in the console.

Additionally, if I am signed into Facebook and have previously authorized my app, clicking the link will "log" me in and redirect me back to my index page immediately, and print the cookie in my console. 

I can see that `Set-Cookie` is sent in both cases. 

Even more strangely, and what makes me think this might be a Rocket bug, is that I can see that the `token` cookie is present in my browser (clicking the little info button in the URL bar), even though it was never printed in the console.

Any thoughts? Thanks!

**Comment by SergioBenitez** [comment]
I cannot reproduce what you describe. The [cookies](https://github.com/SergioBenitez/Rocket/tree/b76a1ef8b99d8f374c405591f27221e6e878d85f/examples/cookies) example does nearly exactly what you've posted. The tests in that example confirm that everything works as expected. I've also manually confirmed the results and have tested that using private cookies also works as expected.

Please provide a failing test case.

**Comment by kardeiz** [comment]
Thanks for your response!

After digging into this some more, I realized that it was being caused by the fact that there was a remote domain in the string of redirects (here facebook.com). And after some lucky googling, I found out that this is actually a bug in (just) Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=696204. I confirmed that it works fine in Firefox.

For now I am setting `same_site` to `Lax`, which resolves the issue, but shouldn't really be necessary.

Speaking of which, would it be possible for rocket to re-export `cookie::SameSite`. `set_same_site` takes a `SameSite`, and is a public function on `Cookie`, so it seems to make sense to re-export it as well. Right now I am bringing my own `cookie` crate, but keeping versions equal could be a hassle.

Closing now. Thanks!
